images/github-runners-manager/manager.py

```diff
@@ -0,0 +1,107 @@
+        for workflow in workflows:
+            for workflow_run in workflow.get_runs(status='queued'):
+                for job in workflow_run.jobs():
+                    if [value for value in args.runner_label if value in job.raw_data['labels']]:
+                        runner_name = f"ggml-runner-{workflow.id}-{job.id}-{workflow_run.event}-{int(time.time())}"
+
+                        print(f"TRIGGERING {runner_name} for workflow_name={workflow.name}")
+                        runner_dir = f'/runners/{runner_name}'
+                        os.makedirs(runner_dir)
+                        os.chmod(runner_dir, 0o755)
+                        os.chown(runner_dir, 1000, 1000)
+
+                        # Get a JIT runner config
+                        jitrequest = {
+                            'name': runner_name,
+                            'runner_group_id': 1,  # FIXME what to put here
+                            'labels': ["self-hosted", "X64", "linux", *args.runner_label],
+                            'work_folder': "/tmp/github-runner/_work"
+                        }
+                        response = requests.post(
+                            f" https://api.github.com/repos/{args.repo}/actions/runners/generate-jitconfig",
+                            headers={
+                                'Authorization': f'Bearer {args.token}',
+                                'X-GitHub-Api-Version': "2022-11-28"
+                            },
+                            json=jitrequest)
+                        if response.status_code != 201:
+                            print(f"invalid JIT response code: {response.status_code}\n    {response.text}")
+                            continue
+                        jitconfig = response.json()
+
+                        # start the worker in its container and wait for finish
+                        print(
+                            f"Running job runner id={jitconfig['runner']['id']} os={jitconfig['runner']['os']} labels={[value['name'] for value in jitconfig['runner']['labels']]}")
+                        try:
+                            client.containers.run("ggml-github-runner", jitconfig['encoded_jit_config'],
+                                                  entrypoint="/entrypoint.sh",
+                                                  name=runner_name,
+                                                  runtime="nvidia",
+                                                  device_requests=[
+                                                      DeviceRequest(device_ids=["all"],
+                                                                    capabilities=[['gpu']])],
+                                                  user='1000:1000',
+                                                  auto_remove=True,
+                                                  tmpfs={
+                                                      f'/mnt/runners/{runner_name}/tmp': 'size=8G,uid=1000'
```

Comment:
I'm not sure if `/mnt/runners/{runner_name}/tmp` will really be used. Please note the what's being specified here is the path inside the created container, so I believe it should be `/tmp/github-runner` in this case (but ideally you should mount the whole `/tmp` as tmpfs)

If we mount the whole `/tmp` as tmpfs, then in the entrypoint you can use `mkdir -p /tmp/github-runner/_work` to create all folder recursively.